release.sh

```diff
@@ -655,6 +655,23 @@ function publish_artifacts() {
   banner "New release published successfully"
 }
 
+# Sets the github release with the highest semver to 'latest'
+function set_latest_to_highest_semver() {
+  # List latest release
+  local releases # don't combine with the line below, or $? will be 0
+  # Support tags in two formats
+  # - knative-v1.0.0
+  # - v1.0.0
+  releases="$(hub_tool -p release | cut -d '-' -f2)"
```

Comment:
This will fail if `set -e` is set. It should be set.

Also, storing array in a regular variable isn't a good idea.

Reply:
This (minus the `-p`) is copy/pasted from https://github.com/knative/hack/blob/main/release.sh#L220 which has been working....

Reply:
I'm not referring to `-p`.

You can't check for `[[ $? -eq 0 ]]` unless `set +e` is active. Otherwise, every failed command exit the script (as it should).

Reply:
I think this could be simplified to:

```bash
local last_version
last_version="$(hub_tool -p release | cut -d'-' -f2 | grep '^v[0-9]\+\.[0-9]\+\.[0-9]\+$'| sort -r -V | head -1)"
```

Reply:
yes, but not as readable imo. technically we can "simplify" lots of this and make it a one liner monster :P

Reply:
Bash isn't readable. Writing invalid Bash, just to be readable, is not a good idea.

See this article: https://stackoverflow.com/a/17602863/844449

Reply:
how is this invalid bash?

Reply:
oh, because you are saying array in a var is bad practice... I wouldn't call it invalid but i see your point

Reply:
invalid bash == one that has bugs in it